numpy/core/shape_base.py

```diff
@@ -276,7 +290,7 @@ def vstack(tup):
     if not overrides.ARRAY_FUNCTION_ENABLED:
         # raise warning if necessary
         _arrays_for_stack_dispatcher(tup, stacklevel=2)
-    return _nx.concatenate([atleast_2d(_m) for _m in tup], 0)
+    return _concatenate([_atleast_2d(_m) for _m in tup], 0)
```

Comment:
This code is strange: why loop over the arrays when `atleast_2d` does that as well? This means one can reduce the overhead in your example by a factor 3 with zero effort or impact!

This of course separate from the very simple change in `atleast_2d` to use `np.array(ary, copy=False, subok=True, ndmin=2)`...

Reply:
p.s. Though in my more ideal world, `atleast_2d` would just call `np.array(...)` only when `.shape` and `.reshape` were not present and otherwise just rely on those. But that's a separate issue!

Reply:
If I recall correctly, the `atleast_*d` functions have different semantics than `np.array(..., ndim=*)`